code/web/src/modules/user/MyProducts.js

```diff
@@ -0,0 +1,155 @@
+                    <th>Delivered On</th>
+                    <th style={{ textAlign: 'center' }} >Kept</th>
+                  </tr>
+                </thead>
+
+                {/*This is where we'll map through all state user products and return the tr:*/}
+                {/*this.userProducts.map(product => {
+                  return (
+                    <tr key={product.id} { !this.state.viewAll && !product.kept && styles={{ visibility: 'hidden' }} }>
+                      <td>
+                        <img src={product.image} alt='product image' />
+                      </td>
+                      <td>
+                        { product.name }
+                      </td>
+                      <td>
+                        { product.description }
+                      </td>
+                      <td>
+                        { product.deliveredDate }
+                      </td>
+                      <td style={{ textAlign: 'center' }}>
+                        {product.kept && <p>⭐️</p>}
+                      </td>
+                    </tr>
+                  )
+                })*/}
+
+                <tbody>
+                  <tr key={666} styles={{ display: this.state.viewAll && kept ? 'block' : 'none' }}>
+                    <td>
+                    <img src='https://hips.hearstapps.com/countryliving.cdnds.net/17/47/2048x1365/gallery-1511194376-cavachon-puppy-christmas.jpg?resize=100:*' alt='profile image' />
+
+                    </td>
+                    <td>
+                      { 'pants' }
+                    </td>
+                    <td>
+                      { 'two legs' }
+                    </td>
+                    <td>
+                      {'11/22/3333'}
+                    </td>
+                    <td style={{ textAlign: 'center' }}>
+                      {kept && <p>⭐️</p>}
+                    </td>
+                  </tr>
+                  <tr key={667} styles={{ display: this.state.viewAll && kept2 ? 'block' : 'none' }}>
+                    <td>
+                    <img src='https://hips.hearstapps.com/countryliving.cdnds.net/17/47/2048x1365/gallery-1511194376-cavachon-puppy-christmas.jpg?resize=100:*' alt='profile image' />
+
+                    </td>
+                    <td>
+                      { 'shirt' }
+                    </td>
+                    <td>
+                      { 'two arms' }
+                    </td>
+                    <td>
+                      {'11/22/3337'}
+                    </td>
+                    <td style={{ textAlign: 'center' }}>
+                      {kept2 && <p>⭐️</p>}
+                    </td>
+                  </tr>
+                </tbody>
+              </table>
+            </GridCell>
+          </Grid>
+        </div>
+      </>
+    )
+  }
+}
+
+// Component Properties
+MyProducts.propTypes = {
+  user: PropTypes.object.isRequired,
+  logout: PropTypes.func.isRequired
+}
+
+// Component State
+function productsState(state) {
```

Comment:
I think is as it should be? I did this, but would appreciate it if someone would double check it.

Reply:
Are you specifically talking about the Redux hookup here with the connect and this `mapStateToProps` function called `productsState`? It looks right to me. I think we have to add the api call in either something that stores the user products in the store or in a `componentDidMount` of this specific component. The latter is my thought since the user only ever sees it here and doesn't interact with the actual data of their 'history of delivered products' in any way.